src/Actions/CartItem/CreateCartItem.php

```diff
@@ -26,34 +30,36 @@ public function performAction(): mixed
     {
         $this->data['amount'] ??= 1;
 
-        $product = null;
-        if ($productId = data_get($this->data, 'product_id')) {
-            $product = Product::query()
-                ->whereKey($productId)
-                ->first(['vat_rate_id', 'name']);
-            $this->data['vat_rate_id'] ??= $product?->vat_rate_id;
-            $this->data['name'] ??= $product?->name;
-        }
+        $cartItem = app(CartItem::class, ['attributes' => $this->data]);
+        $cartItem->save();
+
+        return $cartItem->fresh();
+    }
 
-        $cart = resolve_static(Cart::class, 'query')
+    protected function prepareForValidation(): void
+    {
+        $this->cart = resolve_static(Cart::class, 'query')
```

Comment:
if cart does not exist an exception is thrown, property of type Cart cannot be null.